### PR TITLE
DMP-806 - v42 and v43 SQL updates

### PR DESCRIPTION
**Pull request by cmagowan-hmcts**
https://tools.hmcts.net/jira/browse/DMP-806

I have introduced some missing entity classes, but i have not done all the correct joins to objects, as i felt that was out of scope of this ticket. (e.g. a column called cas_id should be linked to a CourtCaseEntity, but is just an Integer for now.

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/darts/common/entity/base/CreatedModifiedBaseEntity.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+@MappedSuperclass
+@Getter
+@Setter
+public class CreatedModifiedBaseEntity {
+
+    @CreationTimestamp
+    @Column(name = "created_ts")
+    private OffsetDateTime createdDateTime;
+
+    @ManyToOne
```
Comment (richardbull) [inline src/main/java/uk/gov/hmcts/darts/common/entity/base/CreatedModifiedBaseEntity.java:24]: Think we should have `fetch = FetchType.LAZY` here

File: src/main/java/uk/gov/hmcts/darts/common/entity/base/CreatedModifiedBaseEntity.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    @JoinColumn(name = "created_by", referencedColumnName = "usr_id")
+    private UserAccountEntity createdBy;
+
+    @UpdateTimestamp
+    @Column(name = "last_modified_ts")
+    private OffsetDateTime lastModifiedDateTime;
+
+    @ManyToOne
```
Comment (richardbull) [inline src/main/java/uk/gov/hmcts/darts/common/entity/base/CreatedModifiedBaseEntity.java:32]: Think we should have `fetch = FetchType.LAZY` here

File: src/main/java/uk/gov/hmcts/darts/common/entity/base/CreatedModifiedBaseEntity.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+@Getter
+@Setter
+public class CreatedModifiedBaseEntity {
+
+    @CreationTimestamp
+    @Column(name = "created_ts")
+    private OffsetDateTime createdDateTime;
+
+    @ManyToOne
+    @JoinColumn(name = "created_by", referencedColumnName = "usr_id")
```
Comment (richardbull) [inline src/main/java/uk/gov/hmcts/darts/common/entity/base/CreatedModifiedBaseEntity.java:25]: I don't think we need to have the `referencedColumnName = "usr_id"` for the join

File: src/main/java/uk/gov/hmcts/darts/common/entity/base/CreatedModifiedBaseEntity.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    private UserAccountEntity createdBy;
+
+    @UpdateTimestamp
+    @Column(name = "last_modified_ts")
+    private OffsetDateTime lastModifiedDateTime;
+
+    @ManyToOne
+    @JoinColumn(name = "last_modified_by", referencedColumnName = "usr_id")
```
Comment (richardbull) [inline src/main/java/uk/gov/hmcts/darts/common/entity/base/CreatedModifiedBaseEntity.java:33]: I don't think we need to have the referencedColumnName = "usr_id" for the join

File: src/main/java/uk/gov/hmcts/darts/common/entity/base/CreatedModifiedBaseEntity.java [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+import org.hibernate.annotations.CreationTimestamp;
+import org.hibernate.annotations.UpdateTimestamp;
+import uk.gov.hmcts.darts.common.entity.UserAccountEntity;
+
+import java.time.OffsetDateTime;
+
+@MappedSuperclass
+@Getter
+@Setter
+public class CreatedModifiedBaseEntity {
```
Comment (richardbull) [inline src/main/java/uk/gov/hmcts/darts/common/entity/base/CreatedModifiedBaseEntity.java:18]: FYI Looks like Barry will be reinstating the `version integer not null` column soon, so we can pull that column from the `VersionedEntity` class to be our sensible base class. Now that the entities are becoming more uniformly presented.

File: src/main/java/uk/gov/hmcts/darts/common/entity/base/CreatedModifiedBaseEntity.java (hunk 2 above)
Comment (cmagowan-hmcts) [reply src/main/java/uk/gov/hmcts/darts/common/entity/base/CreatedModifiedBaseEntity.java]: added

File: src/main/java/uk/gov/hmcts/darts/common/entity/base/CreatedModifiedBaseEntity.java (hunk 1 above)
Comment (cmagowan-hmcts) [reply src/main/java/uk/gov/hmcts/darts/common/entity/base/CreatedModifiedBaseEntity.java]: done

**Review by richardbull — APPROVED** [review]

**Pull request merged**